src/components/Stepper/src/Stepper.vue

```diff
@@ -124,22 +155,57 @@ export default {
 				return;
 			}
 			const incrementBy = 1;
-			/**
-			 * updated stepper value
-			 * @property {number}
-			 */
-			this.$emit('stepper:update', this.value + incrementBy);
+			this.emitUpdate(this.value + incrementBy);
 		},
 		decrement() {
 			if (!Number.isNaN(this.minVal) && this.value <= this.minVal) {
 				return;
 			}
 			const decrementBy = 1;
+			this.emitUpdate(this.value - decrementBy);
+		},
+
+		triggerManualInput() {
+			this.manualValue = this.value;
+			this.isSettingManualValue = true;
+			this.$nextTick(() => {
+				this.$refs.manualInput.focus();
+				this.$refs.manualInput.select();
+			});
+		},
+
+		commitManualValue(event) {
+			event?.preventDefault?.();
+			event?.stopPropagation?.();
+
+			const newValue = Number.parseInt(this.manualValue, 10);
```

Comment:
since `<input type="number">` lets characters like `.` and `e` get entered we should change this to:
```
const newValue = Math.round(Number.parseFloat(this.manualValue, 10));
```
which correctly handles those inputs, unlike `Number.parseInt`. An alternative solution would be to prevent people from entering `.` or `e` into the input by adding an `input` or `change` event listener onto the `input` element and do something like:
```js
allowCharacter(event) {
  return /\d/.test(event.key); // return false if non-digit character
}
```

Reply:
Good suggestion, I'll do that.